docs/Technical-Documentation/Architecture-Decision-Record/016-zap-waiver.md

```diff
@@ -22,6 +22,21 @@ When investigating, we sought out information on what other, modern ACF systems
 
 While we also investigated and developed technical solutions utilizing either React-app-rewired or Next.js, we decided that the departure from our existing stack was not a worthwhile investment and could lead to other complications down the road.
 
+### 2024 Update
+
+Another instance of this alert appeared in the Webinspect scans for January/February 2024, after the introduction of Kibana to our stack
+
+`HTML5: Misconfigured Content Security Policy` - `Recommendation: Remove the unsafe-eval and unsafe-inline values from the CSP directive values.`
+
+Referring to the following directives in our nginx config
+```
+set $CSP "${CSP}script-src 'self' 'unsafe-eval' 'unsafe-inline' http://{{env "KIBANA_BASE_URL"}}:5601;";
+set $CSP "${CSP}script-src-elem 'self' 'unsafe-inline' http://{{env "KIBANA_BASE_URL"}}:5601;";
+set $CSP "${CSP}script-src-attr 'self' 'unsafe-inline' http://{{env "KIBANA_BASE_URL"}}:5601;";
+```
+
+Kibana performs ajax requests from its frontend to scripts hosted on the server in order to provide styling and javascript functionality - in order to serve Kibana from behind a proxy, which is a requirement with cloud.gov, these `unsafe-inline` and `unsafe-eval` directives are required. The CSP exceptions are fairly low risk as they are limited to artifacts within the Kibana domain.
+
```

Comment:
how do we determine "they are limited to artifacts within the Kibana domain"? I think to be able to resolve this, we will need to move adding the unsafe-inline to location file and reset the CSP header, otherwise the unsafe-inline will be applied to all urls and is not safe. an example to add header in location is (here)[https://docs.hypernode.com/hypernode-platform/nginx/how-to-add-custom-http-headers-to-a-file-or-location.html]

Reply:
The decision is to accept the waiver and reconsider removing unsafe-inline later when we update frontend